twa

```diff
@@ -118,31 +122,50 @@ function stage_0_has_https {
   # First, just probe the domain to see if it listens on 443.
   if ! nc -z -w2 "${domain}" 443 2>/dev/null; then
     FATAL "expected port 443 to be open, but it isn't"
-    no_https=1 ; return
+    no_https=1
```

Comment:
The `return` here should probably be kept in -- `FATAL` means that `twa` has flunked the site unequivocally, and so subsequent checks don't even matter until the `FATAL` one is resolved.

Reply:
Similarly, the `FATAL` lines below should also have a `return` immediately after.

Reply:
Hm. I removed the `return` because on one of the websites I checked, which had an expired SSL cert, this test failed and reported the 443 port to be closed.

Reply:
That's interesting (and possibly a bug); do you have a site I can attempt to repro with?

Reply:
I observed this with `piwo.informatyka.org.`

Reply:
Hmm, I can't reproduce that -- `twa -v` against that site shows me that `nc` connects to port `443`. Maybe it was intermittent?